File: test/data/wrap-translation/js/test_en.js
// NO WRAP:
var xxx = require('xxx');

//
var test = {
  'yyy': 'Text',
  'zzz': true
};

// NO WRAP:
if (test['xxx_1'] == 'text_1') {
  // WRAP:
  x = 'Text';
}

// NO WRAP:
if (x != 'text_1') {
  // WRAP:
  x = 'Text';
}

// NO WRAP:
if (test['xxx_1'] == 'text_1' || test['xxx_2'] === 'text_2' || test['xxx_1'] != 'text_1' || test['xxx_2'] !== 'text_2') {
  // WRAP:
  x = 'Text';
}

// NO WRAP, WRAP
x = (x == 'xxx' ? 'text_1' : 'text_2');

// NO WRAP:
x = bar('text') == baz('text')
x = bar('text') === baz('text')
x = bar('text') != baz('text')
x = bar('text') !== baz('text')
x = bar('text') == baz('text') && bar('text_2') == baz('text_2')

// NO WRAP:
x = test['xxx_1']['text_2'];

// NO WRAP, WRAP
x = $("select[name='type']");
x = x.find('#xxx');
x = $("select[name='type']").text('text_1').find('#' + 'xxx').append("xxx").text('text_1');

// WRAP
$(function(){
  x = 'text_1';
});

// NO WRAP:
x = ".xxx";
x = "#xxx";

// NO WRAP:
x = "?app=filemanager&ajax=true&fileRequired=false";
x = "&mode=getfolder&time=";

// WRAP:
x = "q?xxx";
x = "q&xxx";

// NO WRAP:
x = "http://xxx.com";
x = "mailto:xxx@xxx.com";

// NO WRAP:
x = "-x";
x = "x";

// NO WRAP:
$('body').on('click', '#xxx', function () {
  // WRAP:
  x = 'Text';
});

x = x.append({
  // WRAP:
  text: 'Text'
});

File: test/data/wrap-translation/lua/test_en.lua
-- test

-- NO WRAP
local xxx = require 'xxx'
local xxx = require("xxx")
dofile "xxx"
dofile ('xxx')

-- WRAP
io.write("Hello word!")

-- NO WRAP
x = xxx["xxx"]
x = xxx["xxx"]['yyy']

-- NO WRAP
if x == "xxx" or y ~= [[zzz]] then
  -- WRAP
  x = [[xxx]]
else
  -- WRAP
  x = [[yyy]]
end

-- NO WRAP
x = bar('text') == baz('text')
x = bar('text') ~= baz('text')
x = bar('text') == baz('text') and bar('text_2') == baz('text_2')

-- NO WRAP
foo([[xxx]])
x = a.b.c.foo([[xxx]])

-- NO WRAP, WRAP
local xxx = foo("xxx", {
  Key_1 = 'text',
  Key_2 = 'text'
})

-- NO WRAP
error("xxx\nyyy\n")
error("xxx\n"..err)

-- NO WRAP, WRAP
local xxx = {
  ["yyy"] = {
    ['zzz'] = [[text]]
  }
}

-- NO WRAP:
x = ".xxx";
x = "#xxx";

-- NO WRAP:
x = "?app=filemanager&ajax=true&fileRequired=false";
x = "&mode=getfolder&time=";

-- WRAP:
x = "q?xxx";
x = "q&xxx";

-- NO WRAP:
x = "http://xxx.com";
x = "mailto:xxx@xxx.com";

-- NO WRAP:
x = "-x";
x = "x";

-- NO WRAP:
local SQLstring = [[SELECT ]] .. escape(columns) .. [[ FROM ]]
SQLstring = SQLstring .. [[ TABLE ]]
xxx.SQLstring = SQLstring .. [[ TABLE ]]
xxx.yyy.SQLstring = SQLstring .. [[ TABLE ]]
xxx[yyy].SQLstring = [[SELECT ]]
xxx = {SQLstring = "auxiliary"}

-- NO WRAP:
local function getfieldnames (headers)
  local disposition_hdr = headers["content-disposition"]
  local attrs = {}
  if disposition_hdr then
    gsub(disposition_hdr, ';%s*([^%s=]+)="(.-)"', function(attr, val)
	   attrs[attr] = val
         end)
  else
    error("Error processing multipart/form-data."..
          "\nMissing content-disposition header")
  end
  return attrs.name, attrs.filename
end
